### PR TITLE
docs(meshservice): modify targetref

### PR DESCRIPTION
**Pull request by slonka**
I'm struggling a bit, maybe @michaelbeaumont can tell me what else needs to be changed and how he envisioned this.

Should I go through every policy and change MeshService to be compatible with the new form ([basically adding namespace and modifying the name](https://github.com/kumahq/kuma/blob/e3b1dc7ae89ee5272a561c7fd03737684ea039c1/docs/madr/decisions/046-meshservice-policy-matching.md#meshservice) - or should I always use `labels`?)? Or should I add deprecation notice and an example showing the new way?

I started changing federation guide but waiting for some feedback first.

closes https://github.com/kumahq/kuma-website/issues/1892

rendered: https://deploy-preview-1909--kuma.netlify.app/docs/dev/policies/targetref/

Did you sign your commit? [Instructions](https://github.com/kumahq/.github/blob/main/CONTRIBUTING.md#sign-your-commits) yes

Have you read [Contributing guidelines](https://github.com/kumahq/.github/blob/main/CONTRIBUTING.md)? yes

### CODE REVIEW

**Comment by lobkovilya** [comment]
@slonka does it cover https://github.com/kumahq/kuma-website/issues/1962?

File: app/_src/guides/federate.md [hunk 1]
```diff
@@ -137,6 +138,33 @@ spec:
...
+echo "apiVersion: kuma.io/v1alpha1
+kind: MeshCircuitBreaker
+metadata:
+  name: demo-app-to-redis
+  namespace: {{site.mesh_namespace}}
+  labels:
+    kuma.io/mesh: default
+spec:
+  targetRef:
+    kind: MeshService
```
Comment (lobkovilya) [inline app/_src/guides/federate.md:153]: this looks like a mistake, we can't use `kind: MeshService` with `name` and `namespace` in the top-level targetRef

**Comment by slonka** [comment]
> @slonka does it cover https://github.com/kumahq/kuma-website/issues/1962?

@lobkovilya not really but I guess it should since we're already touching a bunch of stuff

File: app/_src/guides/federate.md (hunk 1 above)
Comment (slonka) [reply app/_src/guides/federate.md:153]: It's still in draft, let me look into that.

**Comment by michaelbeaumont** [comment]
@slonka @lobkovilya I'm closing this since it's mostly covered by https://github.com/kumahq/kuma-website/pull/1956

**Pull request closed without merging**